### PR TITLE
[no ticket] poll until we don't get a StorageException

### PR DESCRIPTION
**Pull request by jaycarlton**
Instead of waiting a fixed interval, keep trying to access the bucket until it works or we run out of attempts.

### CODE REVIEW

File: integration/src/main/java/scripts/testscripts/ControlledGcsBucketLifecycle.java [hunk 1]
```diff
@@ -193,11 +194,21 @@ public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspace
     logger.info("Added {} as a reader to workspace {}", reader.userEmail, getWorkspaceId());
 
     // TODO(PF-643): this should happen inside WSM.
-    logger.info("Waiting 30s for permissions to propagate");
-    TimeUnit.SECONDS.sleep(30);
-
-    // Second user can now read the blob
-    Blob readerRetrievedFile = readerStorageClient.get(blobId);
+    int numTries = 20;
```
Comment (zloery) [inline integration/src/main/java/scripts/testscripts/ControlledGcsBucketLifecycle.java:197]: Can you pull this into a utility or something similar? We have a few other tests which use a similar pattern of "add user to workspace, wait for cloud permissions" that could reuse this logic

**Review by zloery — COMMENTED** [review]
I filed PF-854 yesterday when the test was failing, feel free to attach that here for tracking/credit

File: integration/src/main/java/scripts/utils/ClientTestUtils.java [hunk 2]
```diff
@@ -223,6 +226,37 @@ public static boolean jobIsRunning(JobReport jobReport) {
...
+   * sleep for a specific duration, and retry up to a fixed number of times. This structure
+   * is useful for situations where we are waiting on a cloud IAM permission change to take effect.
+   * @param supplier - code returning the result or throwing an exception
+   * @param numTries - number of times to retry the operation
+   * @param sleepDuration - sleep time between attempts
+   * @param <T> - type of result
+   * @return - result from supplier, the first time it doesn't throw
+   * @throws InterruptedException
+   */
+  public static <T> T getWithRetryOnException(Supplier<T> supplier, int numTries, Duration sleepDuration) throws InterruptedException {
```
Comment (ddietterich) [inline integration/src/main/java/scripts/utils/ClientTestUtils.java:240]: Nice. Thanks for this!

**Review by ddietterich — APPROVED** [review]
lgtm

File: integration/src/main/java/scripts/utils/ClientTestUtils.java [hunk 3]
```diff
@@ -223,6 +226,37 @@ public static boolean jobIsRunning(JobReport jobReport) {
...
 
+  /**
+   * Get a result from a call that might throw an exception. Treat the exception as retryable,
+   * sleep for a specific duration, and retry up to a fixed number of times. This structure
+   * is useful for situations where we are waiting on a cloud IAM permission change to take effect.
+   * @param supplier - code returning the result or throwing an exception
+   * @param numTries - number of times to retry the operation
+   * @param sleepDuration - sleep time between attempts
+   * @param <T> - type of result
+   * @return - result from supplier, the first time it doesn't throw
```
Comment (zloery) [inline integration/src/main/java/scripts/utils/ClientTestUtils.java:237]: nit: or null if the operation does not succeed within numTries

**Review by zloery — APPROVED** [review]

**Pull request merged**